### ISSUE TITLE
Reopen: ng-xi18n tool compiles TypeScript files

### ISSUE THREAD

**Issue by Martin-Wegner** [angular/angular#13567]
Sorry that I must reopen issue #12305 in this way. But @vicb closed the issue although it has never been fixed. I asked @vicb multiple times (here and in gitter) to reopen the issue but he never responded...

**I'm submitting a ...**

```
[x] bug report
```

**Current behavior**
Having a Webpack 2 project (https://angular.io/docs/ts/latest/guide/webpack.html -> no Angular CLI) and running the `ng-xi18n` tool will compile all TypeScript files. The result is a `src` directory full with `.js`, `.js.map` and `.metadata.json` files.

**Expected behavior**
The `ng-xi18n` tool should only generate the messages file.

**What is the motivation / use case for changing the behavior?**
A `src` directory without `.js`, `.js.map` and `.metadata.json` files.

**Please tell us about your environment:**
Linux
- **Angular version:** 2.3.1
- **Browser:** All
- **Language:** TypeScript 2.0.10
- **Node (for AoT issues):** 6.9.2

**Current workaround:**

```
"prei18n": "rimraf ./temp && shx mv ./tsconfig.json ./tsconfig.bak.json && shx mv ./tsconfig.i18n.json ./tsconfig.json",
"i18n": "ng-xi18n && shx mv ./messages.xlf ./src/app/locale",
"posti18n": "shx mv ./tsconfig.json ./tsconfig.i18n.json && shx mv ./tsconfig.bak.json ./tsconfig.json && rimraf ./temp"
```

`tsconfig.i18n.json` contains `"outDir": "temp",`.

**Comment by vicb** [comment]
You'd better add a repo or I will close again ;)

**Comment by Martin-Wegner** [comment]
What exactly do you need? Take a webpack 2 project as described here https://angular.io/docs/ts/latest/guide/webpack.html with a littble bit i18n and run `./node_modules/.bin/ng-xi18n` and you will see the problem...

**Comment by vicb** [comment]
```
git clone git@github.com:angular/angular2-seed.git
npm i
./node_modules/.bin/ng-xi18n 
```

no generated .js, ...

**Comment by Martin-Wegner** [comment]
@vicb what are talking about??? In the angular2-seed project you see the `outDir` property in the `tsconfig.json` file. This was mentioned in #12305.

Your example shows the problem!

`npm run build` and you will see `main.bundle.js` and `main.map` in the `dist` directory. This is the webpack2 result.

`./node_modules/.bin/ng-xi18n` and you will see a `src` directory with `.js`, `.js.map` and `.metadata.json` in the `dist` direcory. This ist `not` the webpack2 result you want and this files are not part of the wbepack2 result!!!

In a webpack2 project you should not set the `outDir` property in the `tsconfig.json` file! So remove this property and then your `src` directory will be full with `.js`, `.js.map` and `.metadata.json` files.

So please reopen #12305 or this issue!

**Comment by vicb** [comment]
`dist/` indeed contain generated files

**Comment by vicb** [comment]
The issue here is that we call `tsc.main()` in the `extract_i18n.ts` - we should actually have a stripped down version of this function copied over to `extract_18n.ts` and remove the emit part.

**Comment by Martin-Wegner** [comment]
Thank you @vicb !

**Comment by ocombe** [comment]
By the way, this means that using AoT (ngc) + xi18n compiles all the files 2 times right ?
Would it be possible to compile only once and execute both xi18n generation and AoT at the same time ?

**Comment by vicb** [comment]
@ocombe not sure to understand your question. `ngx-i18n` currently compiles the ts files and this should be fixed. `ngc` compiles them once.
what should be done is that `ngx-i18n` should not compile the files. extraction and compilation stays different processes.

**Comment by ocombe** [comment]
Ok we agree then :)
Right now it compiles twice, and it should only compile when using ngc.

**Comment by aroget** [comment]
+1

**Comment by napalm684** [comment]
What release did this make it into?

**Comment by ocombe** [comment]
In 4.0.0
